messaging/src/main/java/org/axonframework/eventhandling/SequencingEventHandlingComponent.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventhandling;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.eventhandling.async.SequencingPolicy;
+import org.axonframework.messaging.Message;
+import org.axonframework.messaging.MessageStream;
+import org.axonframework.messaging.QualifiedName;
+import org.axonframework.messaging.unitofwork.ProcessingContext;
+
+import java.util.Optional;
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Decorator implementation of {@link EventHandlingComponent} that uses a configurable {@link SequencingPolicy} as
+ * fallback to determine the sequence identifier for events, while delegating all other operations to an underlying
+ * {@link EventHandlingComponent}.
+ * <p>
+ * This component first delegates sequence identification to the underlying component. If the delegate returns
+ * {@link Optional#empty()}, then the configured sequencing policy is used as a fallback to determine the sequence
+ * identifier. This allows for customizing the sequencing behavior without overriding existing sequencing logic.
+ * <p>
+ *
+ * @author Mateusz Nowak
+ * @since 5.0.0
+ * @see SequencingPolicy
+ * @see EventHandlingComponent
+ */
+public class SequencingEventHandlingComponent implements EventHandlingComponent {
```

Comment:
Currently, is not used, but I assume to do not duplicate the logic of SequencingPolicy usage the decoration approach is desirable. In the following PRs I will implement some new policies like bases on property using Converter.